numba/tests/test_np_functions.py

```diff
@@ -771,20 +775,144 @@ def test_delete_exceptions(self):
             str(raises.exception)
         )
 
+        with self.assertRaises(IndexError) as raises:
+            cfunc([1, 2], 3)
+        self.assertIn(
+            'obj must be less than the len(arr)',
+            str(raises.exception),
+        )
+
         with self.assertRaises(TypingError) as raises:
             cfunc(2, 3)
         self.assertIn(
             'arr must be either an Array or a Sequence',
             str(raises.exception)
         )
 
+    def test_delete3_basic(self):
```

Comment:
Maybe add those above stated failures in the review as test cases when they're fixed.